packages/stark-ui/src/modules/pretty-print/components/pretty-print.component.spec.ts

```diff
@@ -270,910 +151,191 @@ describe("PrettyPrintComponent", () => {
-				expect(formattedData).toContain("CreateNewDoc");
-				expect(formattedData).toContain("menuitem");
-				expect(formattedData).not.toBe(formattedJsonData);
-
-				const preElement: HTMLPreElement | null = <HTMLPreElement>hostFixture.nativeElement.querySelector("pre");
-				expect(preElement).not.toBeNull();
-				expect(preElement.innerHTML).toContain('<code class="language-json">');
-				expect(preElement.innerHTML).toContain('<span class="token string">"file"</span>');
-				expect(preElement.innerHTML).toContain('<span class="token property">"onclick"</span>');
-			});
-
-			it("should remove highlighting from the already highlighted JSON data when the enableHighlighting is set to FALSE", () => {
-				hostComponent.enableHighlighting = false;
-				hostFixture.detectChanges();
-
-				const formattedData: string = component.prettyString;
-
-				expect(formattedData).not.toContain("class='language-json'");
-				expect(formattedData).not.toContain(classTokenProperty);
-				expect(formattedData).not.toContain("<span");
-				expect(formattedData).toContain("menu");
-				expect(formattedData).toContain("CreateNewDoc");
-				expect(formattedData).toContain("menuitem");
-				expect(formattedData).toBe(formattedJsonData);
-
-				const preElement: HTMLPreElement | null = <HTMLPreElement>hostFixture.nativeElement.querySelector("pre");
-				expect(preElement).not.toBeNull();
-				expect(preElement.innerHTML).not.toContain('<code class="language-json">');
-				expect(preElement.innerHTML).toContain('"menuitem": [');
-				expect(preElement.innerHTML).toContain('"onclick": "CreateNewDoc()"');
-			});
-
-			it("should not highlight invalid JSON data", () => {
-				const invalidRawJsonData: string = rawJsonData.replace(":", "oops");
-
-				hostComponent.data = invalidRawJsonData;
-				hostComponent.format = "json";
-				hostComponent.enableHighlighting = true;
-				hostFixture.detectChanges();
-
-				const formattedData: string = component.prettyString;
-
-				expect(formattedData).not.toContain("class='language-json'");
-				expect(formattedData).not.toContain(classTokenProperty);
-				expect(formattedData).not.toContain("<span");
-				expect(formattedData).toContain("menu");
-				expect(formattedData).toContain("CreateNewDoc");
-				expect(formattedData).toContain("menuitem");
-				expect(formattedData).toBe(invalidRawJsonData);
-
-				const preElement: HTMLPreElement | null = <HTMLPreElement>hostFixture.nativeElement.querySelector("pre");
-				expect(preElement).not.toBeNull();
-				expect(preElement.innerHTML).not.toContain('<code class="language-json">');
-				expect(preElement.innerHTML).toContain('"menu"oops');
-			});
+		it("should highlight when success and enableHighlighting", () => {
+			mockPrettyPrintService.format.and.returnValue(of(highloghtedHtmlData));
+
+			hostComponent.format = "html";
+			hostComponent.data = rawHtmlData;
+			hostComponent.enableHighlighting = true;
+			hostFixture.detectChanges();
+			
```

Comment:
Since we would remove the `beforeAll` at the beginning, we should adapt this code as well.
It might be adapted like this:

```suggestion
		it("should highlight when success and enableHighlighting", fakeAsync(() => {
			const starkPrettyPrintService = new StarkPrettyPrintServiceImpl(component.logger);
			mockPrettyPrintService.format.and.returnValue(starkPrettyPrintService.format(rawHtmlData, "html", true));
			
			hostComponent.format = "html";
			hostComponent.data = rawHtmlData;
			hostComponent.enableHighlighting = true;
			hostFixture.detectChanges();
			tick(500);

```